### ISSUE TITLE
Shallow

### ISSUE THREAD

**Issue by eddyerburgh** [vuejs/vue-test-utils#9]
I've been playing around with shallow and I think it'll be very difficult to add support for all CSS selectors in `find`, `contains` and `is`.

I just did a speed test of shallow in it's current state vs mount using this test:

```js
it('toggles class when div is clicked', () => {
    const wrapper = shallow(SpeedTestComponent)
    expect(wrapper.hasClass('active')).to.equal(false)
    wrapper.trigger('click')
    expect(wrapper.hasClass('active')).to.equal(true)
    wrapper.trigger('click')
    expect(wrapper.hasClass('active')).to.equal(false)
})
```
## Results

| Tests        | shallow  (render in s)         | mount (render in s)   |
| ------------- |:-------------:| -----:|
|1   | 2.6 | 3.8 |
| 10      | 2. 7     |   3.6 |
| 100 | 3.7     |   5.4 |
| 1000 | 86     |    88  |

The setup was using `mocha-webpack` with `jsdom` in the mount tests.

I ran each test 5 times and took the average.

As you can see, there's a small constant difference between shallow and mount. (also running 1000 tests in 1 file with my setup took a very long time!)

Maybe we should not develop `shallow`.

Or have a `shallow` method that stubs all child components and then calls `mount`.

What do you guys think?

**Comment by yyx990803** [comment]
That's interesting (and good to see real numbers)! Indeed looks like the perf advantage isn't significant enough to warrant maintaining the two modes in parallel.

I agree that we should drop it for now and focus on other areas.

**Comment by blocka** [comment]
As i understand it, the purpose of shallow rendering is not for speed but
to be able to test the invariants of your component (ie, it should render a
Foo with props x and y)

**Comment by yyx990803** [comment]
To clarify - I think we should drop the VDOM-based rendering mode, but implement shallow (with stubbed child components) on top of current mounting mechanism.

**Comment by eddyerburgh** [comment]
This has been implemented in b07e38c2a13b072166f9e5c8160791377356562f
